app/models/folio/holdings.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module Folio
+  ###
+  #  Winnows down the entire holdings to just what was requested by the user
+  ###
+  class Holdings
+    # @param [Request] request the users request
+    # @param [String] instance_id the FOLIO instance_id (a UUID)
+    def initialize(request, instance_id)
+      @request = request
+      @instance_id = instance_id
+    end
+
+    # @return [Array<OpenStruct>] a list of every holding in the requested library/location with the given barcodes
+    def where(barcodes: [])
+      return [] if barcodes.empty?
+
+      all.select do |item|
+        barcodes.include?(item.barcode)
+      end
+    end
+
+    # @return [Array<OpenStruct>] a list of every holding in the requested library/location
+    def all
+      @all ||= items_in_location.map do |item|
+        item.with_status(@request.item_status(item.barcode))
+      end
+    end
+
+    def single_checked_out_item?
+      all.one? &&
+        all.first.current_location.try(:code) == 'CHECKEDOUT'
```

Comment:
Where is `current_location` defined? Should we use the `checked_out?` accessor you have below?

Reply:
Yes, I think this was a bad rebase prior to https://github.com/sul-dlss/sul-requests/commit/1f25a5573f0ecf0bf654e554d1204dd2e45901e4#diff-b9a80913142bd85afc2319ad52bd074b495c84e7e2e9acf11dba1eac8305fd07L36